dotCMS/src/main/java/com/dotcms/graphql/AbstractGraphQLHttpServlet.java

```diff
@@ -0,0 +1,446 @@
+        }
+
+        String operationName = null;
+        final Optional<Part> operationNameItem = getFileItem(fileItems, "operationName");
+        if (operationNameItem.isPresent()) {
+            operationName = new String(ByteStreams.toByteArray(operationNameItem.get().getInputStream())).trim();
+        }
+
+        graphQLRequest = new GraphQLRequest(query, variables, operationName);
+        return graphQLRequest;
+    }
+
+    private void mapMultipartVariables(GraphQLRequest request,
+                                       Map<String, List<String>> variablesMap,
+                                       Map<String, List<Part>> fileItems) {
+        Map<String, Object> variables = request.getVariables();
+
+        variablesMap.forEach((partName, objectPaths) -> {
+            Part part = getFileItem(fileItems, partName)
+                    .orElseThrow(() -> new RuntimeException("unable to find part name " +
+                            partName +
+                            " as referenced in the variables map"));
+
+            objectPaths.forEach(objectPath -> VariableMapper.mapVariable(objectPath, variables, part));
+        });
+    }
+
+    public void addListener(GraphQLServletListener servletListener) {
+        configuration.add(servletListener);
+    }
+
+    public void removeListener(GraphQLServletListener servletListener) {
+        configuration.remove(servletListener);
+    }
+
+    @Override
+    public String[] getQueries() {
+        return configuration.getInvocationInputFactory().getSchemaProvider().getSchema().getQueryType().getFieldDefinitions().stream().map(GraphQLFieldDefinition::getName).toArray(String[]::new);
+    }
+
+    @Override
+    public String[] getMutations() {
+        return configuration.getInvocationInputFactory().getSchemaProvider().getSchema().getMutationType().getFieldDefinitions().stream().map(GraphQLFieldDefinition::getName).toArray(String[]::new);
+    }
+
+    @Override
+    public String executeQuery(String query) {
+        try {
+            return configuration.getObjectMapper().serializeResultAsJson(configuration.getQueryInvoker().query(configuration.getInvocationInputFactory().create(new GraphQLRequest(query, new HashMap<>(), null))));
+        } catch (Exception e) {
+            return e.getMessage();
+        }
+    }
+
+    private void doRequestAsync(HttpServletRequest request, HttpServletResponse response, HttpRequestHandler handler) {
+        if (configuration.isAsyncServletModeEnabled()) {
+            AsyncContext asyncContext = request.startAsync();
+            HttpServletRequest asyncRequest = (HttpServletRequest) asyncContext.getRequest();
+            HttpServletResponse asyncResponse = (HttpServletResponse) asyncContext.getResponse();
+            new Thread(() -> doRequest(asyncRequest, asyncResponse, handler, asyncContext)).start();
```

Comment:
this should be handle by the DotConcurrentFactory instead of creating Thread in any request